src/tool/hpcrun/audit/auditor.c

```diff
@@ -94,6 +94,9 @@
 #error "GOT resolver index for the host architecture is unknown"
 #endif
 
+#define IS_VDSO  true
+#define NOT_VDSO false
```

Comment:
I'd prefer to keep these written as their more obvious `true` and `false`, unless there might be other options (in which case I would prefer an `enum` of flag bits). In my experience macros tend to complicate debugging and code readability, and so should only be used sparingly when no other option is available.

Reply:
I hate seeing true/false arguments because they don't make any sense without looking at the a function's argument definition. How about this:

`static const bool vdso_load_module = true;`

pass as argument: vdso_load_module, ! vdso_load_module

Reply:
I would argue the `static const` is worse, I can't tell at a glance that `vdso_load_module` is immutable and it acts like a real variable (`&vdso_load_module` makes no sense but is not malformed). You usually have to check the function's declaration for any real work anyway, and gdb will give you the argument names as part of the backtrace.

The other option I agree with is an enum of flag bits, it has the syntax additions of a macro and the debuggability of a standard type (gdb will break it out for you: `#0  foo (x=(X_1 | X_2)) at x.c:10`).
```c
enum audit_open_flags {
  AO_NONE = 0x0,
  AO_VDSO = 0x1,
};
...
static void hook_open(uintptr_t* cookie, struct link_map* map, enum audit_open_flags flags) {
  // is_vdso = flags & AO_VDSO
...
```
Usage: `AO_VDSO`, `AO_NONE`.

Reply:
Of course, all of this is predicated that you can examine the code under the debugger, which you can't. I want to write it for readability, which is the only way that one can debug auditor code. 

I will switch to these flags, which read a lot like what I originally had.